arm-customer-insights/2017-01-01/swagger/customer-insights.json

```diff
@@ -4646,9 +4764,57 @@
         "primaryParticipantProfilePropertyName": {
           "type": "string",
           "description": "The primary participant property name for an interaction ,This is used to logically represent the agent of the interaction, Specify the participant name here from ParticipantName."
+        },
+        "dataSources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataSource"
+          },
+          "description": "This is specific to interactions modeled as activities. Data sources are used to determine where data is stored and also in precedence rules."
+        },
+        "defaultDataSourceId": {
+          "type": "string",
+          "description": "Default data source is specifically used in cases where data source is not specified in an instance."
+        },
+        "isActivity": {
+          "type": "boolean",
+          "description": "An interaction can be tagged as an activity only during create. This enables the interaction to be editable and can enable merging of properties from multiple data sources based on precedence, which is defined at a link level."
         }
       }
     },
+    "DataSource": {
+      "properties": {
+        "dataSourceType": {
+          "type": "string",
+          "enum": [
+            "ConnectorMapping",
+            "LinkInteraction",
+            "SystemDefault"
+          ],
+          "x-ms-enum": {
+            "name": "DataSourceType",
+            "modelAsString": false
+          },
+          "description": "The data source type."
```

Comment:
Marked as readonly. Also changed to modelAsString since there maybe more types in the future.